driver/controller.go

```diff
@@ -194,13 +195,22 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	if snapshot != nil && vol != nil && volumeReq.SnapshotID != "" && int(volumeReq.SizeGigaBytes) > int(snapshot.SizeGigaBytes) && int(volumeReq.SizeGigaBytes) > 1 {
-		log.Infof("volume %s will be resize from %d to %d", vol.ID, int(snapshot.SizeGigaBytes), int(volumeReq.SizeGigaBytes))
-		_, resp, err := d.storageActions.Resize(ctx, vol.ID, int(volumeReq.SizeGigaBytes), volumeReq.Region)
-		log.WithField("response", resp).Info("volume was resized")
+	if snapshot != nil && int(volumeReq.SizeGigaBytes) > int(snapshot.SizeGigaBytes) && int(volumeReq.SizeGigaBytes) > 1 {
```

Comment:
How would that occur ? Snapshots are created from other volume claims and volume claims create volumes with a integer size between 1 to 10 000 GB. Thus, we are always handling integers here if I understand correctly.

Reply:
Snapshots can also be created outside of Kubernetes. Specifically, given the API allows to specify a float, then it should be possible for a user to also create a snapshot through the API using a float size.

I haven't tested it though, so that's just my reasoning based on type inspections.